src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedArr = '';
+    let pseudoSeparator = separator;
+
+    if (separator === null) {
+      pseudoSeparator = 'null';
+    }
+
+    for (const n of this) {
+      let item = n + '';
+
+      if (typeof n === 'undefined' || n === null) {
+        item = '';
+      }
+
+      if (this.lastIndexOf(n) === this.length - 1) {
+        joinedArr += item;
+        break;
+      }
+      joinedArr += item + pseudoSeparator;
```

Comment:
looks too complicated, can you add item before your check and the decide in if if you should add the separator? cause you're adding item anyways. then you can remove your `break`)